lib/virtfs/context.rb

```diff
@@ -1,4 +1,10 @@
 module VirtFS
+  # FS-specific settings under which FS calls occur.
+  #
+  # VirtFS maps an independent context instance to each
+  # mounted FS instance, and internally switches to it before
+  # dispatching target FS calls. This class implements the
```

Comment:
There is not an independent context per mounted FS instance. There is a context per thread group of the process - so normally, there will only be one context per process.

Due to the nature of constants in ruby, VirtFS _activation_ must be a process-while operation. When a given application requires a process to have multiple _views_ of the current FS state, multiple contexts can be used. By dividing the process into multiple thread-groups, each thread-group can have it's own view of the current FS state. For example, one group might only see the native filesystem, while others can see additional filesystems, or totally different filesystems. Basing contexts on thread-groups, as opposed to individual threads, permits multi-threaded filesystem access when necessary.